### PR TITLE
Fix caching of variable attributes

### LINKED ISSUE #109: casadi load_model() api replaces parameter values in expressions
When loading variable metadata from a cached model, the ```load_model()``` method does three calls of the ```variable_metadata_function()```. In the process, symbolic expressions that contain parameters with known values are resolved into a numeric value prematurely, so that parameter values cannot be overwritten later. Effectively, cached models behave as though the ```replace_parameter_values``` compiler option is true.

### PR DESCRIPTION
**Pull request by jackvreeken**
- [x] Add (previously failing) test cases for both commits

### CODE REVIEW

File: src/pymoca/backends/casadi/api.py [hunk 1]
```diff
@@ -257,39 +257,18 @@ def load_model(model_folder: str, model_name: str, compiler_options: Dict[str, s
...
                 variable = variable_dict[d['name']]
                 for j, tmp in enumerate(CASADI_ATTRIBUTES):
-                    if not getattr(variable, tmp).is_regular():
-                        if (not isinstance(metadata[key][i, j], ca.DM)
-                            and ca.depends_on(metadata[key][i, j], parameter_vector)):
-                            setattr(variable, tmp, metadata[key][i, j])
+                    if ca.depends_on(metadata[key][i, j], parameter_vector):
+                        setattr(variable, tmp, metadata[key][i, j])
+                    else:
+                        setattr(variable, tmp, independent_metadata[key][i, j])
```
Comment (jbaayen) [inline src/pymoca/backends/casadi/api.py]: So what happens now is that parameters of the form

`parameter Real a = 1.0;`

get their numerical value exposed.

OTOH parameters like

`parameter Real b = a;`

do not. The original intent was to produce the value `1.0` when accessing `b` as well.

So this is an API change which will have ramifications for RTC-Tools as well as some applications. For instance, now, any calculations in `pre()` using parameter values may fail if there is an in-between assignment anywhere.

Comment (jackvreeken) [reply src/pymoca/backends/casadi/api.py]: A cached model should have exactly the same attributes as a freshly parsed model. This PR is addressing that discrepancy. If the behavior of a freshly parsed model should be different, then we should discuss and address that separately.

Comment (jbaayen) [reply src/pymoca/backends/casadi/api.py]: I thought the behaviour was the same at some point in time, but perhaps I am mistaken.

I agree about consistency but bear in mind that this change may cause breakage in RTC models which expect pymoca to behave like JModelica in this regard.

Comment (jackvreeken) [reply src/pymoca/backends/casadi/api.py]: There were no checks in place for the correctness of the `variable_metadata_function`, and also none for the additional steps in `load_model()`. It's therefore hard to say whether they were ever equal.

**Pull request merged**